coding-standards/css.md

```diff
@@ -1,11 +1,348 @@
+}
+```
+
+**Good**
+
+```scss
+.element {
+  margin: 10px 0 (@variable * 2) 10px;
+}
+```
+
+### Comments
+
+Code is written and maintained by people. Ensure your code is descriptive, well-commented, and approachable by others. Great code comments convey context or purpose. Do not simply reiterate a component or class name.
+
+Be sure to write in complete sentences for larger comments and succinct phrases for general notes.
+
+Comments have to be used to group specific CSS for a particular section of the page.
+
+**Bad**
+
+```css
+/* Modal header */
+.modal-header {
+  ...
+}
+```
+
+**Good**
+
+```css
+/* Wrapping element for .modal-title and .modal-close */
+.modal-header {
+  ...
+}
+```
+
+### Class Names
+Follow BEM convention: https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/#the-bem-naming-convention
+- Keep classes lowercase and use dashes (not underscores or camelCase). Dashes serve as natural breaks in related class (e.g., <span style="color: #ff9900;">`.btn`</span> and <span style="color: #ff9900;">`.btn-danger`</span>).
+- Avoid excessive and arbitrary shorthand notation. <span style="color: #ff9900;">`.btn`</span> is useful for *button*, but <span style="color: #ff9900;">`.s`</span> doesn't mean anything.
+- Keep classes as short and succinct as possible.
+- Use meaningful names; use structural or purposeful names over presentational.
+- Prefix classes based on the closest parent or base class.
+- Use <span style="color: #ff9900;">`.js-*`</span> classes to denote behavior (as opposed to style) or classes that are added by JavaScript, but keep these classes out of your CSS.
+- Follow modularity and specificity: https://docs.ckan.org/en/2.9/contributing/css.html#modularity-and-specificity
+
+It's also useful to apply many of these same rules when creating Sass and Less variable names.
+
+**Bad**
+
+```css
+.t { ... }
+.red { ... }
+.header { ... }
+```
+
+**Good**
+
+```css
+.tweet { ... }
+.important { ... }
+.tweet-header { ... }
+```
+
+### Selectors
+
+- Use classes over a generic element tag for optimum rendering performance.
+- Avoid using several attribute selectors (e.g., <span style="color: #ff9900;">`[class^="..."]`</span>) on commonly occurring components. Browser performance is known to be impacted by these.
+- Keep selectors short and strive to limit the number of elements in each selector to three.
+- Scope classes to the closest parent **only** when necessary (e.g., when not using prefixed classes).
+
+**Bad**
+
+```css
+span { ... }
+.page-container #stream .stream-item .tweet .tweet-header .username { ... }
+.avatar { ... }
+```
+
+**Good**
+
+```css
+.avatar { ... }
+.tweet-header .username { ... }
+.tweet .avatar { ... }
+```
+
+Reference: [http://codeguide.co/#css](http://codeguide.co/#css)
+
+# Enforcing Tool and Config
+
+- Use Stylelint for linting. [https://stylelint.io/user-guide/get-started](https://stylelint.io/user-guide/get-started)
```

Comment:
Do we need any configuration for stylelint to ensure our rules are followed?
Or the default stylelint rules should take care?

Reply:
Have specified some rules too.
In order to specify properties order we need this plugin though.
https://github.com/hudochenkov/stylelint-order